admin/class-customizer.php

```diff
@@ -106,149 +125,105 @@ public function breadcrumbs_blog_remove_active_cb() {
-			'wpseo_titles[breadcrumbs-searchprefix]', array(
-				'default'   => '',
-				'type'      => 'option',
-				'transport' => 'refresh',
-			)
+		$index        = 'breadcrumbs-searchprefix';
+		$control_args = array(
+			'label' => __( 'Prefix for search result pages:', 'wordpress-seo' ),
 		);
 
-		$this->wp_customize->add_control(
-			new WP_Customize_Control(
-				$this->wp_customize, 'wpseo-breadcrumbs-searchprefix', array(
-					'label'    => __( 'Prefix for search result pages:', 'wordpress-seo' ),
-					'type'     => 'text',
-					'section'  => 'wpseo_breadcrumbs_customizer_section',
-					'settings' => 'wpseo_titles[breadcrumbs-searchprefix]',
-					'context'  => '',
-				)
-			)
-		);
+		$this->add_setting_and_control( $index, $control_args );
 	}
 
 	/**
 	 * Adds the breadcrumb 404 prefix text field
 	 */
 	private function breadcrumbs_404_setting() {
-		$this->wp_customize->add_setting(
-			'wpseo_titles[breadcrumbs-404crumb]', array(
-				'default'   => '',
-				'type'      => 'option',
-				'transport' => 'refresh',
-			)
+		$index        = 'breadcrumbs-404crumb';
+		$control_args = array(
+			'label' => __( 'Breadcrumb for 404 pages:', 'wordpress-seo' ),
 		);
 
-		$this->wp_customize->add_control(
-			new WP_Customize_Control(
-				$this->wp_customize, 'wpseo-breadcrumbs-404crumb', array(
-					'label'    => __( 'Breadcrumb for 404 pages:', 'wordpress-seo' ),
-					'type'     => 'text',
-					'section'  => 'wpseo_breadcrumbs_customizer_section',
-					'settings' => 'wpseo_titles[breadcrumbs-404crumb]',
-					'context'  => '',
-				)
-			)
-		);
+		$this->add_setting_and_control( $index, $control_args );
+	}
+
+	/**
+	 * Adds the customizer setting and control.
+	 *
+	 * @param string $index         Array key index to use for the customizer setting.
+	 * @param array  $control_args  Customizer control object arguments.
+	 *                              Only those different from the default need to be passed.
+	 * @param string $id            Optional. Customizer control object ID.
+	 *                              Will default to 'wpseo-' . $index.
+	 * @param array  $settings_args Optional. Customizer setting arguments.
+	 *                              Only those different from the default need to be passed.
+	 */
+	private function add_setting_and_control( $index, $control_args, $id = null, $settings_args = array() ) {
+		$setting                  = sprintf( $this->setting_template, $index );
+		$control_args             = array_merge( $this->default_control_args, $control_args );
+		$control_args['settings'] = $setting;
+
+		if ( empty( $settings_args ) ) {
+			$settings_args = $this->default_setting_args;
+		}
+		else {
+			$settings_args = array_merge( $this->default_setting_args, $settings_args );
+		}
+
+		if ( ! isset( $id ) ) {
+			$id = 'wpseo-' . $index;
+		}
+
+		$this->wp_customize->add_setting( $setting, $settings_args );
+
+		$control_obj = new WP_Customize_Control( $this->wp_customize, $id, $control_args );
```

Comment:
Please don't abbreviate variable names (`$control_obj`). Note that `[...]args` is an exception to this. 

Perhaps consider renaming the variable to `$control`.

Reply:
Good suggestion.